### ISSUE TITLE
Intrinsic fir.sqrt.f32.f32 is multiply defined

### ISSUE THREAD

**Issue by sscalpone** [flang-compiler/f18-llvm-project#289]
The symbol `fir.sqrt.f32.f32 `is given global linkage, but it appears to be generated in all object files that use `SQRT`.

E.g. from BLAS:
```
[  6%] Linking Fortran executable ../../bin/xblat1s
/home/sjs/work/lapack/lapack/BLAS/SRC/srotg.f:109: multiple definition of `fir.sqrt.f32.f32'
```

```
% cat sq.f
      SUBROUTINE SROTF(SA,SB)
      INTRINSIC SQRT
      REAL SA,SB
      SB = SQRT(SA)
      RETURN
      END
```
```
% flang -c sq.f
warning: overriding the module target triple with x86_64-unknown-linux-gnu [-Woverride-module]
1 warning generated.
```
```
% nm sq.o
0000000000000020 T fir.sqrt.f32.f32
                 U __mth_i_sqrt
0000000000000000 T _QPsrotf
```

**Comment by jeanPerier** [comment]
@schweitzpgi do you know how to specify internal linkage on a `mlir::FuncOp` ?

I could not find documentation for this and that looks unimplemented to me in mlir based on the following comment:
https://github.com/llvm/llvm-project/blob/9db53a182705ac1f652c6ee375735bea5539272c/mlir/lib/Conversion/StandardToLLVM/StandardToLLVM.cpp#L1084-L1088

I tried to use the mlir symbol visibility (`func.setVisibility(mlir::FuncOp::Visibility::Private);`), but that does not set the llvm linkage to internal (I am not sure it is intended to though).

@sscalpone, I can disable intrinsic outlining as a temporary workaround (they will still be generated for unrestricted intrinsics passed as arguments, the workaround there would take more time).

**Comment by jeanPerier** [comment]
The temporary workaround was pushed in #290, I keep this open until we actually fix the underlying linkage issue.

**Comment by sscalpone** [comment]
The code is working now with the checked-in compiler (with the workaround).

**Comment by jeanPerier** [comment]
Closed in fir-dev by #306.
nm output is now:
```
0000000000000020 t fir.sqrt.f32.f32
                 U __mth_i_sqrt
0000000000000000 T _QPsrotf
```

Note: I had to revert mlir commit 4589dd924dfc43c846652b85825e291af0d7428a to get this program compiling, but that is a different issue coming from the latest rebase.
